components/motor/gpio/setup.go

```diff
@@ -182,15 +183,40 @@ func createNewMotor(
 	if err != nil {
 		return nil, err
 	}
+
 	if motorConfig.Encoder != "" {
+		basic := m.(*Motor)
 		e, err := encoder.FromDependencies(deps, motorConfig.Encoder)
 		if err != nil {
 			return nil, err
 		}
 
-		m, err = WrapMotorWithEncoder(ctx, e, cfg, *motorConfig, m, logger)
+		props, err := e.Properties(context.Background(), nil)
 		if err != nil {
-			return nil, err
+			return nil, errors.New("cannot get encoder properties")
+		}
+		if !props.TicksCountSupported {
+			return nil,
+				encoder.NewEncodedMotorPositionTypeUnsupportedError(props)
+		}
+
+		single, isSingle := e.(*single.Encoder)
+		if isSingle {
+			single.AttachDirectionalAwareness(basic)
+			logger.CInfo(ctx, "direction attached to single encoder from encoded motor")
+		}
```

Comment:
I like this but it seems to be breaking functionality. when I move this into `WrapMotorWithEncoder` and `setupMotorWithControls`, and add the directionMoving functions to those respective motor types, everything seems to work as expected

Reply:
will investigate more to see if there's a way to keep it as you have it here

Reply:
looks like it's actually a locking issue. if I remove a couple of the locks around `SetPower` and `DirectionMoving` it works but I'm not sure if that's unsafe yet will have to test more